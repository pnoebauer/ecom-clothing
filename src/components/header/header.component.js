import React from 'react';
import { Link } from 'react-router-dom';
import { auth } from '../../firebase/firebase.utils';
import { connect } from 'react-redux';

import './header.styles.scss';

import { ReactComponent as Logo } from '../../assets/logo.svg';
import CartIcon from '../cart-icon/cart-icon.component';
import CartDropdown from '../cart-dropdown/cart-dropdown.component.js';
import { selectCurrentUser } from '../../redux/user/user.selectors';
import { selectCartHidden } from '../../redux/cart/cart.selectors';


const Header = ({ currentUser, hideCartDropdown }) => (
	<div className='header'>
		<Link className='logo-container' to='/'>
			<Logo className='logo'/>
		</Link>
		<div className='options'>
			<Link className='option' to='/shop'> 
				SHOP
			</Link>
			<Link className='option' to='/shop'> 
				CONTACT
			</Link>
			{
				currentUser ?
				<div className='option' onClick={() => auth.signOut()} >SIGN OUT</div>
				:
				<Link className='option' to='/signin'>SIGN IN</Link>
			}
			<CartIcon/>
		</div>
		{
			hideCartDropdown ?
			null 
			:
			<CartDropdown />
		}
	</div>
);

const mapStateToProps = state => ({
	currentUser: selectCurrentUser(state),
	hideCartDropdown: selectCartHidden(state)
});

export default connect(mapStateToProps)(Header);