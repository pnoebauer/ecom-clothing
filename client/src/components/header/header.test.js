import React from 'react';
import {shallow} from 'enzyme';

import {Header} from './header.component';
import CartDropdown from '../cart-dropdown/cart-dropdown.component';

import {
	HeaderContainer,
	LogoContainer,
	OptionsContainer,
	OptionContainer,
} from './header.styles';

describe('Header component', () => {
	let wrapper;
	let mockSignOutStart;

	beforeEach(() => {
		mockSignOutStart = jest.fn();

		const mockProps = {
			hideCartDropdown: true,
			currentUser: {
				uid: '123',
			},
			signOutStart: mockSignOutStart,
		};

		wrapper = shallow(<Header {...mockProps} />);
	});

	test('render Header component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('if currentUser is present', () => {
		test('render sign out link', () => {
			expect(wrapper.find(OptionContainer).at(2).text()).toEqual('SIGN OUT');
		});

		test('call signOutStart method when link is clicked', () => {
			wrapper.find(OptionContainer).at(2).simulate('click');

			expect(mockSignOutStart).toHaveBeenCalled();
		});
	});

	describe('if currentUser is null', () => {
		test('render sign in link', () => {
			const mockProps = {
				hideCartDropdown: true,
				currentUser: null,
				signOutStart: mockSignOutStart,
			};

			const newWrapper = shallow(<Header {...mockProps} />);

			expect(newWrapper.find(OptionContainer).at(2).text()).toEqual('SIGN IN');
		});
	});

	describe('if hideCartDropdown is true', () => {
		test('do not render CartDropdown', () => {
			expect(wrapper.exists(CartDropdown)).toEqual(false);
		});
	});

	describe('if currentUser is null', () => {
		test('render CartDropdown', () => {
			const mockProps = {
				hideCartDropdown: false,
				currentUser: null,
				signOutStart: mockSignOutStart,
			};

			const newWrapper = shallow(<Header {...mockProps} />);

			expect(newWrapper.exists(CartDropdown)).toEqual(true);
		});
	});
});
