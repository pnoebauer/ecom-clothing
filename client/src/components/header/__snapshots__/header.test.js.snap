// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render Header component 1`] = `
<headerstyles__HeaderContainer>
  <headerstyles__LogoContainer
    to="/"
  >
    <ForwardRef(SvgLogo)
      className="logo"
    />
  </headerstyles__LogoContainer>
  <headerstyles__OptionsContainer>
    <headerstyles__OptionContainer
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/shop"
    >
      SHOP
    </headerstyles__OptionContainer>
    <headerstyles__OptionContainer
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/contact"
    >
      CONTACT
    </headerstyles__OptionContainer>
    <headerstyles__OptionContainer
      onClick={[MockFunction]}
    >
      SIGN OUT
    </headerstyles__OptionContainer>
    <Connect(CartIcon) />
  </headerstyles__OptionsContainer>
</headerstyles__HeaderContainer>
`;
